src/test/store/profile-view.test.js

```diff
@@ -902,6 +902,51 @@ describe('actions/ProfileView', function () {
       expect(getMarker(markerIndexes[1]).name.includes('b')).toBeTruthy();
     });
 
+    it('filters the markers by searchable unique-string fields', function () {
+      const profile = getProfileWithMarkers([
+        [
+          'a',
+          5,
+          10,
+          {
+            type: 'StringTesting',
+            searchableString: 'searchable cucumber',
+            searchableUniqueString: 'searchable mango',
+            nonSearchableString: 'non-searchable papaya',
+            nonSearchableUniqueString: 'non-searchable onion',
+          },
+        ],
+      ]);
+      const { dispatch, getState } = storeWithProfile(profile);
+
+      expect(
+        selectedThreadSelectors.getSearchFilteredMarkerIndexes(getState())
+      ).toHaveLength(1);
+
+      const getMarker = selectedThreadSelectors.getMarkerGetter(getState());
+      const markerPayload: MixedObject = (getMarker(0).data: any);
+      expect(typeof markerPayload.searchableString).toBe('string');
+      expect(typeof markerPayload.searchableUniqueString).toBe('number');
+      expect(typeof markerPayload.nonSearchableString).toBe('string');
+      expect(typeof markerPayload.nonSearchableUniqueString).toBe('number');
+      function getMarkerIndexesForSearch(searchString) {
```

Comment:
nit: please add an empty line before the function so that it stands out a bit more. I also considered asking you to move it to a setup function or something, but probably not worth it.